src/Components/TestExplorer.fs

```diff
@@ -1060,10 +1060,32 @@ module Interactions =
         )
         |> Promise.ofThenable
 
+    let private filterEscapeCharacter = '\\'
+    let private filterSpecialCharacters = [| '\\'; '('; ')'; '&'; '|'; '='; '!'; '~' |]
+    let private filterSpecialCharactersSet = Set.ofArray filterSpecialCharacters
 
     let private buildFilterExpression (tests: TestItem array) =
+        // Escape any special characters in test names, as per https://github.com/microsoft/vstest/blob/main/docs/filter.md
+        let escapeFilterExpression (str: string) =
+
+            if str.IndexOfAny(filterSpecialCharacters) < 0 then
+                str
+            else
+                let builder = StringBuilder()
+
+                for i = 0 to str.Length do
```

Comment:
hmm, clicked submit and then thought 'hang on, that's one too many characters' :-(